File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFragmentImpl.java
package com.jameskbride.eventbusmvpdemo.network;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jameskbride.eventbusmvpdemo.R;
import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent;
import com.jameskbride.eventbusmvpdemo.network.NetworkErrorViewPresenter.NetworkErrorView;

import javax.inject.Inject;

import static com.jameskbride.eventbusmvpdemo.network.NetworkErrorViewFragment.NETWORK_REQUEST;

public class NetworkErrorViewFragmentImpl implements NetworkErrorView{

    private NetworkErrorViewPresenter presenter;
    private NetworkErrorViewFragment networkErrorViewFragment;

    @Inject
    public NetworkErrorViewFragmentImpl(NetworkErrorViewPresenter presenter) {
        this.presenter = presenter;
    }

    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState, final NetworkErrorViewFragment networkErrorViewFragment) {
        this.networkErrorViewFragment = networkErrorViewFragment;
        View view = inflater.inflate(R.layout.network_error, container);
        presenter.view = this;

        view.findViewById(R.id.retry_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                NetworkRequestEvent networkRequestEvent = (NetworkRequestEvent) networkErrorViewFragment.getArguments().getSerializable(NETWORK_REQUEST);
                presenter.retry(networkRequestEvent);
            }
        });

        return view;
    }

    public void onResume(NetworkErrorViewFragment networkErrorViewFragment) {
    }

    public void onPause(NetworkErrorViewFragment networkErrorViewFragment) {
    }

    @Override
    public void dismiss() {
        networkErrorViewFragment.dismiss();
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFragmentImpl.kt
package com.jameskbride.eventbusmvpdemo.network

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import com.jameskbride.eventbusmvpdemo.R
import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent
import com.jameskbride.eventbusmvpdemo.network.NetworkErrorViewFragment.Companion.NETWORK_REQUEST
import javax.inject.Inject


class NetworkErrorViewFragmentImpl @Inject constructor(val presenter: NetworkErrorViewPresenter) : NetworkErrorView {
    lateinit var networkErrorViewFragment: NetworkErrorViewFragment

    fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?, networkErrorViewFragment: NetworkErrorViewFragment): View {
        this.networkErrorViewFragment = networkErrorViewFragment
        val view = inflater!!.inflate(R.layout.network_error, container)
        presenter.view = this

        view.findViewById<Button>(R.id.retry_button).setOnClickListener {view: View? ->
            val networkRequestEvent = networkErrorViewFragment.arguments?.getSerializable(NETWORK_REQUEST) as NetworkRequestEvent
            presenter.retry(networkRequestEvent)
        }

        return view
    }

    fun onResume(networkErrorViewFragment: NetworkErrorViewFragment) {}

    fun onPause(networkErrorViewFragment: NetworkErrorViewFragment) {}

    override fun dismiss() {
        networkErrorViewFragment.dismiss()
    }
}